framework/include/base/MooseObjectWarehouseBase.h

```diff
@@ -539,6 +550,56 @@ MooseObjectWarehouseBase<T>::updateVariableDependencyHelper(std::set<MooseVariab
   }
 }
 
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateMatPropDependency(std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveObjects(tid))
+    updateMatPropDependencyHelper(needed_mat_props, _all_objects[tid]);
+}
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateBlockMatPropDependency(SubdomainID id, std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveBlockObjects(id, tid))
+    updateMatPropDependencyHelper(needed_mat_props, getActiveBlockObjects(id, tid));
+}
+
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateBoundaryMatPropDependency(std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveBoundaryObjects(tid))
+  {
+    typename std::map<BoundaryID, std::vector<MooseSharedPointer<T> > >::const_iterator it;
+    for (it = _active_boundary_objects[tid].begin(); it != _active_boundary_objects[tid].end(); ++it)
+      updateMatPropDependencyHelper(needed_mat_props, it->second);
+  }
+}
+
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateBoundaryMatPropDependency(BoundaryID id, std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveBoundaryObjects(id, tid))
+    updateMatPropDependencyHelper(needed_mat_props, getActiveBoundaryObjects(id, tid));
+}
+
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateMatPropDependencyHelper(std::set<unsigned int> & needed_mat_props, const std::vector<MooseSharedPointer<T> > & objects)
+{
+  for (typename std::vector<MooseSharedPointer<T> > ::const_iterator it = objects.begin(); it != objects.end(); ++it)
```

Comment:
`auto` should work here and in other places (I think). Never tried it in a place where typename is required.